pydantic_ai_slim/pydantic_ai/ag_ui.py

```diff
@@ -378,10 +378,9 @@ async def run_ag_ui(
                 yield encoder.encode(event)
 
         if on_complete is not None and run.result is not None:
-            if _utils.is_async_callable(on_complete):
-                await on_complete(run.result)
-            else:
-                await _utils.run_in_executor(on_complete, run.result)
+            result = on_complete(run.result)
+            if asyncio.iscoroutine(result):
+                await result
```

Comment:
Please revert this change unless it's related to this PR and/or intentional, as this would mean we're no longer running `on_complete` in a thread -- which may be a good thing.